integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/strategy/outbox/OutboxPollingAutomaticIndexingStrategyLifecycleIT.java

```diff
@@ -50,25 +50,25 @@ public void cleanUp() {
 	public void stopWhileOutboxEventsIsBeingProcessed() {
 		SessionFactory sessionFactory = setup();
 		backendMock.verifyExpectationsMet();
+		int size = 1000;
 
 		OrmUtils.withinTransaction( sessionFactory, session -> {
-			for ( int i = 1; i <= 1000; i++ ) {
+			for ( int i = 1; i <= size; i++ ) {
 				IndexedEntity entity = new IndexedEntity();
 				entity.setId( i );
 				entity.setIndexedField( "value for the field" );
 				session.persist( entity );
 			}
 
 			BackendMock.DocumentWorkCallListContext context = backendMock.expectWorks( IndexedEntity.NAME );
-			for ( int i = 1; i <= 1000; i++ ) {
+			for ( int i = 1; i <= size; i++ ) {
 				context.add( i + "", b -> b.field( "indexedField", "value for the field" ) );
 			}
 			context.createdThenExecuted();
 		} );
 
 		// wait for the first call is processed (partial progressing)
 		outboxEventFinder.showAllEventsUpToNow( sessionFactory );
-		backendMock.awaitFirstDocumentWorkCall();
```

Comment:
You are not waiting until at least one event has been processed, here.

Sure there's an assertion below, but that assertion could fail... And we don't want tests to fail randomly just because the test thread was faster than the background processing thread.

Maybe you could add this here:

```suggestion
		backendMock.indexingWorkThreadingExpectations().awaitIndexingAssertions( () -> {
			OrmUtils.withinTransaction( sessionFactory, session -> {
				assertThat(  outboxEventFinder.findOutboxEventIdsNoFilter( session ) ).hasSizeLessThan( size );
			} );
		} );
```